client.go

```diff
@@ -156,11 +156,11 @@ func NewSnowthClient(discover bool, addrs ...string) (*SnowthClient, error) {
 		return nil, err
 	}
 
-	return NewClient(cfg)
+	return NewClient(context.Background(), cfg)
```

Comment:
I might advise using [context.WithCancel](https://pkg.go.dev/context#WithCancel) and adding a field to the SnowthClient type for the ctx cancel function, though I need to read more of the code to determine if that would be worthwhile.